### ISSUE TITLE
GraphQl: filtering on nested collections 

### ISSUE THREAD

**Issue by Siregacvek** [api-platform/core#2119]
Good evening.

I have an issue with filtering on nested collection. Example:
Given this query returns 2 items:
```
{
  products(name: "apple"){
    edges{
      node{
        id
        name
      }
    }
  }
}
```
It works fine and filters a big set of products properly. All the items are ManyToOne to parent entity `ProductList` with id 1.

Then I have 2 issues:
1. When I try to write a query like this:
```
{
  productList(id: "/api/product_lists/1") {
    products(name: "apple") {
      edges {
        node {
          id
          name
        }
      }
    }
  }
}
```
I can't do this, because schema builder doesn't see filter argument for `products`. This would be great if schema builder will be able to use filters for nested collections as they are described in nested entity.

2. Even if I add special filter to `products` property under `ProductList` like this:
 ```
/** 
 * @ApiResource()
 * @ORM\Entity(repositoryClass="App\Repository\ProductListRepository")
 *...
 class ProductList 
	/**
	 * @ApiSubresource()
	 * @ORM\OneToMany(targetEntity="App\Entity\Main\Product", mappedBy="productList")
     * @ApiFilter(SearchFilter::class, properties={"products.name": "ipartial"})
	 */
	private $products;
```
In this case schema builder sees my new filter and I'm able to send the following:
```
{
  productList(id: "/api/product_lists/92") {
    products(products_name: "apple") {
      edges {
        node {
          id
          name
        }
      }
    }
  }
}
```
But this query returns the full set of products.
After quick code review it looks like you ignore filters on nested collections - at least I didn't find a place where Api-Platform process them.

**Comment by Siregacvek** [comment]
I found a way how to do this: under property declaration I have to put filter like this (without of `products` prefix):
```
/** 
* @ApiResource()
* @ORM\Entity(repositoryClass="App\Repository\ProductListRepository")
*...
class ProductList 
   /**
    * @ApiSubresource()
    * @ORM\OneToMany(targetEntity="App\Entity\Main\Product", mappedBy="productList")
    * @ApiFilter(SearchFilter::class, properties={"name": "ipartial"})
    */
   private $products;
```

And then query works:
```
{
  productList(id: "api/product_lists/1") {
        products(name:"apple") {
          edges {
            node {
              name
            }
          }
        }
      }
    }
```

**Comment by alanpoulain** [comment]
Closed in favor of #2149
